test/test_create_observations.js

```diff
@@ -1,84 +1,300 @@
-  'dataArray@iot.count': 2,
-  dataArray: [ [ 20, 1 ],
-    [ 30, 1 ] ]
-}];
-
-const CreateObservationsRequestComponentsDataArrayMismatch = [{
-  Datastream: { '@iot.id': 1 },
-  components: [ 'phenomenonTime', 'result', 'FeatureOfInterest/id' ],
-  'dataArray@iot.count': 2,
-  dataArray: [ [ '2010-12-23T10:20:00-0700', 1 ],
-    [ '2010-12-23T10:21:00-0700', 1 ] ]
-}];
-
-const CreateObservationsRequestNotArray = {
-  Datastream: { '@iot.id': 1 },
-  components: [ 'phenomenonTime', 'result', 'FeatureOfInterest/id' ],
-  'dataArray@iot.count': 2,
-  dataArray: [ [ '2010-12-23T10:20:00-0700', 20, 1 ],
-    [ '2010-12-23T10:21:00-0700', 30, 1 ] ]
-};
-
-describe('CreateObservations tests', () => {
-  it('Valid request body', done => {
-    postError(done, CreateObservationsRequestOneObservation, 501);
-  });
+db().then(models => {
+  const postSuccess = (done, body, expected) => {
 
-  it('Missing Datastream', done => {
-    postError(done, CreateObservationsRequestMissingDatastream, 400,
-              'Missing required field: Datastream');
-  });
+    server.post(endpoint).send(body)
+    .expect(201)
+    .end((err, res) => {
 
-  it('Missing PhenomenonTime', done => {
-    postError(done, CreateObservationsRequestMissingPhenomenonTime, 400,
-              'Missing required field: phenomenonTime');
-  });
+      const getInstance = (url, index) => {
+        if (url === error) {
+          results[index] = { error: url };
+          return Promise.resolve();
+        }
+
+        let re = /.+\((\d+)\)$/;
+        let id = re.exec(url)[1];
+        return new Promise(resolve => {
+          models[observations].findById(id).then(instance => {
+            results[index] = {
+              datastreamId: instance.DatastreamId,
+              result: instance.result,
+              phenomenonTime: instance.phenomenonTime,
+              featureOfInterestId: instance.FeatureOfInterestId
+            };
+            resolve();
+          });
+        });
+      };
+
+      const response = JSON.parse(res.text);
+      let results = new Array(response.length);
+      let i = 0;
+
+      Promise.all(response.map(url => {
+        return getInstance(url, i++);
+      })).then(() => {
+        should.not.exist(err);
+        res.status.should.be.equal(201);
+        for (let k = 0; k < results.length; k++) {
+          expected[k].should.be.deepEqual(results[k]);
+        }
+        done();
+      });
+    });
+  };
+
+  const postError = (done, body, code, message) => {
+    server.post(endpoint).send(body)
+    .expect(code)
+    .end((err, res) => {
+      should.not.exist(err);
+      res.status.should.be.equal(code);
+      message && res.body.message.should.be.equal(message);
+      done();
+    });
+  };
+
+  const CreateObservationsRequest = [{
+    Datastream: { iotId: 1 },
+    components: [ phenomenonTime, result, featureOfInterestId ],
+    'dataArray@iot.count': 4
+  }];
+
+  const CreateObservationsRequestNoFoI = [{
```

Comment:
nit: it seems that we could use some [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) here to avoid copying all the fields for each test case.